### PR TITLE
Add support to delete team member/application stashes.

### PR DESCRIPTION
**Pull request by crast**
* local statisticians can delete members/apps which they just
  created by mistake, *before* submit.
* regionals can also override the delete and additionally delete team
  items which have already been used in a report.

### CODE REVIEW

File: src/resources/assets/js/submission/applications/components.jsx [hunk 1]
```diff
@@ -346,6 +347,43 @@ class _EditCreate extends ApplicationsBase {
...
+        let extraConfirm, spiel
+        if (!currentApp.id) {
+            return
+        } else if (currentApp.meta && currentApp.meta.canDelete) {
+            spiel = <p>Deleting an application will cause the application to be permanently removed.</p>
+        } else if (lookups.user.canOverrideDelete) {
+            extraConfirm = nameRow(currentApp)
+            spiel = (
+                <div>
+                    <p><b>Regional Statisticians</b> - This application cannot be deleted by a statistician.
```
Comment (pdarg) [inline src/resources/assets/js/submission/applications/components.jsx]: will probably have to update this after the soft deletes are complete

File: src/resources/assets/js/submission/core/DeleteWarning.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                </Form>
+            </Panel>
+        )
+    }
+
+    preSubmit(data) {
+        if (!confirm(`Are you SURE you want to delete this ${this.props.noun}?`)) {
+            return // stop action continuing
+        }
+        if (this.props.extraConfirm) {
```
Comment (pdarg) [inline src/resources/assets/js/submission/core/DeleteWarning.jsx]: should this be comparing the confirm values?

File: src/resources/assets/js/submission/team_members/data.js [hunk 3]
```diff
@@ -49,3 +49,7 @@ export const weeklyReportingSave = new LoadingMultiState('team_members/saveWeekl
 export const weeklyReportingData = new InlineBulkWork('team_members/weeklyReporting')
 
 export const messages = new MessageManager('team_members')
+
+export function teamMemberText(teamMember) {
```
Comment (pdarg) [inline src/resources/assets/js/submission/team_members/data.js]: this should probably be aligned to have the same/similar logic for apps/team_members

**Comment by pdarg** [comment]
[Allow deleting applications, team members in submission](https://trello.com/c/ZbZQhaDj/214-allow-deleting-applications-team-members-in-submission)

**Comment by crast** [comment]
[Bug: no option to delete an added team application(by mistake)](https://trello.com/c/b5prmxOn/129-bug-no-option-to-delete-an-added-team-applicationby-mistake)

**Pull request merged**